### PR TITLE
Feature/sdl 0046 logger abstraction

### PR DESCRIPTION
**Pull request by VladSemenyuk**
[Things to note: Pull Requests **must** fix an issue. Discussion about the feature / bug takes place in the issue, discussion of the implementation takes place in the PR. Please also see the [Contributing Guide](https://github.com/smartdevicelink/sdl_core/blob/master/.github/CONTRIBUTING.md) for information on branch naming and the CLA, and the [SmartDeviceLink GitHub Best Practices](https://d83tozu1c8tt6.cloudfront.net/media/resources/SDL_GitHub_BestPractices.pdf) document for more information on how to enter a pull request. Once this PR is ready for review, please request one from @theresalech.

Delete the above section when you've read it.]

Fixes #[issue number]

This PR is **[ready / not ready]** for review.

### Risk
This PR makes **[no / minor / major]** API changes.

### Testing Plan
[Describe how you plan to unit test the changes in this PR]

### Summary
[Summary of PR changes]

##### Breaking Changes
* [Breaking change info]

##### Enhancements
* [Enhancement info]

##### Bug Fixes
* [Bug Fix Info]

### Tasks Remaining:
- [ ] [Task 1]
- [ ] [Task 2]

### CLA
- [ ] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_rpc_plugin.h [hunk 1]
```diff
@@ -42,11 +42,14 @@
 #include "rc_rpc_plugin/rc_capabilities_manager.h"
 #include "rc_rpc_plugin/rc_consent_manager.h"
 #include "rc_rpc_plugin/resource_allocation_manager.h"
+#include "utils/ilogger.h"
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_rpc_plugin.h:45]: Do we need both forward declaration and include?

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_rpc_plugin.h [hunk 2]
```diff
@@ -42,11 +42,14 @@
...
 
 namespace rc_rpc_plugin {
 namespace plugins = application_manager::plugin_manager;
 namespace app_mngr = application_manager;
 
+class Logger;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_rpc_plugin.h:51]: forward declaration in wrong namespace

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_rpc_plugin.h (hunk 1 above)
Comment (VladSemenyuk) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_rpc_plugin.h:45]: no

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_rpc_plugin.h (hunk 2 above)
Comment (VladSemenyuk) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_rpc_plugin.h]: deleted

**Comment by LuxoftAKutsan** [comment]
delivery PR:  https://github.com/smartdevicelink/sdl_core/pull/3472

**Pull request closed without merging**